design-proposals/multiarch-cluster.md

```diff
@@ -0,0 +1,149 @@
+
+**Virt-config**
+
+- Addition of MultiArchitecture feature gate 
+- Addition of defaults for each of the added KubevirtConfiguration fields
+ 
+**Virt-operator**
+
+- The addition of node-labeler initContainer for virt-handler daemonsets no longer performs architecture checks. The logic for determining whether the initContainer’s node-labeller.sh script runs has now been moved to the script itself.
+
+**Virt-launcher**
+
+- Given that the virt-launcher pod runs on a per VM basis, relying on runtime.GOARCH is an acceptable
+
+**Virt-handler**
+
+- Given that the virt-handler runs as a daemonset, relying on runtime.GOARCH is acceptable
+
+
+## API Examples
+
+**Example VM Spec**
+
+```yaml
+apiVersion: kubevirt.io/v1alpha3
+kind: VirtualMachine
+metadata:
+  name: mytestvm
+spec:
+  running: true
+  template:
+    spec:
+      nodeSelector:
+        kubernetes.io/arch: arm64
+      architecture: arm64
+      domain:
+        devices:
+          disks:
+          - name: disk1
+            disk:
+              bus: virtio
+          interfaces:
+          - name: network1
+            bridge: {}
+      networks:
+              - name: network1
+                multus:
+                  networkName: mytestvm/mytestvm
+      volumes:
+           - name: disk1
+             emptyDisk:
+               capacity: 50G
+```
+
+**Example Kubevirt Configuration**
+
+```yaml	
+apiVersion: kubevirt.io/v1
+kind: KubeVirt
+metadata:
+  name: kubevirt
+  namespace: kubevirt
+spec:
+  certificateRotateStrategy: {}
+  configuration:
+    developerConfiguration:
+      featurettes: ["MultiArchitecture"]
+    archConfiguration:
+      arm64:
+        emulatedMachines: ["virt"]
+    defaultVMIArchitecture: "amd64"
+  customizeComponents: {}
+  imagePullPolicy: IfNotPresent
+```
+
+## Scalability
+
+The proposed changes have no anticipated impact on scalability capabilities of the KubeVirt framework
+
+
+## Update/Rollback Compatibility
+
+The only issue as far as rollback compatibility goes would be around the KubeVirt components within k8s no longer being compiled for multi-architecture environments. If a user were to downgrade their deployment of KubeVirt while still running a multi-architecture cluster, there may be deployment issues for the core components but this would be an expected outcome. 
+
+## Functional Testing Approach
+
+Currently, the only area of this proposal that still requires clarity is what direction the KubeVirt maintainers would like to proceed with regarding functional testing. The existing cluster that is stood up as part of the functional tests leverages a helper script from the k8s repo that checks system architecture prior to bringing up a cluster. How to leverage this script but bring up a multi-architecture cluster still needs discussion.
+
+# Implementation Phases
+
+There is no expectation to need to break this into several implementation phases
```

Comment:
I think we can break out the multi-arch image build in  an extra step. Let's make it a two-phase approach: 

 * One PR, code: where we can still test that pods are created in the e2e test for the  expected architecture, despite not being able to start up.
 * Second PR, pushing multi-arch images as part of the release process